src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImplTest.java

```diff
@@ -123,27 +144,147 @@ public void getAllScriptNames_fail() throws Exception, SQLException {
+        .isEqualTo(baseScript);
+  }
+
+  @Test
+  public void getScript_sortingRenderedOnlyWithSortingColumns_success() {
+    scriptManager = new ScriptManagerImpl(scriptRunner, scriptsMap, sortingColumnsMap);
+    String renderedWithSortingColumns =
+        scriptManager.getScript(
+            sqlTemplateRenderer, "default_chunked", ImmutableList.of("SORTING_COLUMN"));
+    assertThat(renderedWithSortingColumns)
+        .isEqualTo(baseScript + "\nORDER BY SORTING_COLUMN ASC NULLS FIRST\n");
+    String renderedWithOutSortingColumns =
+        scriptManager.getScript(sqlTemplateRenderer, "default_chunked", ImmutableList.of());
+    assertThat(renderedWithOutSortingColumns).isEqualTo(baseScript);
   }
 
   @Test
   public void getScript_fail() {
-    scriptManager = new ScriptManagerImpl(scriptRunner, scriptsMap);
+    scriptManager = new ScriptManagerImpl(scriptRunner, scriptsMap, sortingColumnsMap);
     assertThrows(
         IllegalArgumentException.class,
-        () -> scriptManager.getScript(sqlTemplateRenderer, "not_available_name"));
+        () ->
+            scriptManager.getScript(sqlTemplateRenderer, "not_available_name", ImmutableList.of()));
+  }
+
+  private DataFileReader<Record> getReaderFromAvroFile(Path filePath) throws IOException {
+    DatumReader<Record> datumReader = new GenericDatumReader<>();
+    return new DataFileReader<>(filePath.toFile(), datumReader);
+  }
+
+  private void assertRecordEqualsExpected(Record record, Integer id, String timestampUtc) {
+    assertThat(record.get(0)).isEqualTo(id);
+    assertThat(record.get(1)).isEqualTo(Instant.parse(timestampUtc).toEpochMilli());
+  }
+
+  @Test
+  public void executeScript_writeChunked_success() throws Exception {
+    scriptManager = new ScriptManagerImpl(scriptRunner, scriptsMap, sortingColumnsMap);
+    Connection connection = DriverManager.getConnection("jdbc:hsqldb:mem:db_3");
+    Statement baseStmt = connection.createStatement();
+    baseStmt.execute(
+        "CREATE Table TestTable ("
+            + "ID INTEGER,"
+            + "TIMESTAMPS TIMESTAMP(6) WITH TIME ZONE"
+            + ")");
+    // Insert in different orders to test that the later query does apply ORDER BY.
+    for (int i = 16; i > 10; i--) {
+      baseStmt.execute(
+          String.format(
+              "INSERT INTO TestTable VALUES (%d, TIMESTAMP '2008-08-08 20:08:%02d.007000'"
+                  + " AT TIME ZONE INTERVAL '0:00' HOUR TO MINUTE)",
+              i, i + 8));
+    }
+    for (int i = 0; i < 11; i++) {
+      baseStmt.execute(
+          String.format(
+              "INSERT INTO TestTable VALUES (%d, TIMESTAMP '2008-08-08 20:08:%02d.007000'"
+                  + " AT TIME ZONE INTERVAL '0:00' HOUR TO MINUTE)",
+              i, i + 8));
+    }
+    baseStmt.close();
+    connection.commit();
+    ImmutableList<String> expectedFiles =
+        ImmutableList.<String>builder()
+            .add("default_chunked-20080808T200808S007000-20080808T200810S007000_0.avro")
```

Comment:
I'm curious, if there is a specific reason to use builder, instead of ImmutableList.ofI().

Reply:
Builder pattern felt more expandable at the time of writing...